### DISCUSSION TITLE
C for beginners: All 8kyu Codewars problems sorted by "C Primer Plus" chapter

### DISCUSSION THREAD

**Discussion by darrellroot** [codewars/codewars.com#2880] [General]
For the C language, not all 8kyu problems are "easy".  Many require array, string, or dynamic memory management which "introduction to C" students can't handle for at least their first month.

I went through all 180 8kyu C problems and sorted them by the chapter those topics are introduced in Stephen Prata's, "C Primer Plus" book:

https://github.com/darrellroot/Computer-Science-Education-Resources/blob/main/Codewars-Problems-By-Stephen-Prata-Chapter.md

For example, here are the 7 C problems I think a student can complete after reading chapter 2 of the book:

* BASIC: Making Six Toast https://www.codewars.com/kata/5834fec22fb0ba7d080000e8
* Get Nth Even Number https://www.codewars.com/kata/5933a1f8552bc2750a0000ed
* Multiply https://www.codewars.com/kata/50654ddff44f800200000004
* Third Angle of a Triangle https://www.codewars.com/kata/5a023c426975981341000014
* Function 2 - squaring an argument https://www.codewars.com/kata/523b623152af8a30c6000027
* You Can't Code Under Pressure 1 https://www.codewars.com/kata/53ee5429ba190077850011d4
* Beginner Series 2 Clock https://www.codewars.com/kata/55f9bca8ecaa9eac7100004a

The list should be considered "draft" and I welcome pull requests.  Once the lists are a bit better vetted I will generate corresponding Codewars collections.

Share and enjoy!  - Darrell

**Comment by Kacarott** [comment]
This is really neat! One suggestion, you could make a "collection" in Codewars for each chapter, and link to that collection, alongside the kata links. I think that makes it slightly easier for users to train on one kata after another in a row.
